docs/adr/adr-009-telemetry.md

```diff
@@ -0,0 +1,226 @@
+provide, which we can use as well to extract valuable insight on how the network performs in macro view.
+
+### Tooling/Dependencies
+
+#### Golang API/Shim
+
+The decision is to use [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go) for both Metrics and Tracing:
+* Minimal and golang savvy API/shim which gathers years of experience from OpenCensus/OpenMetrics and [CNCF](https://www.cncf.io/)
+* Backends/exporters for all the existing timeseries monitoring DBs, e.g. Prometheus, InfluxDB. As well as tracing backends
+  Jaeger, Uptrace, etc.
+* https://github.com/uptrace/opentelemetry-go-extra/tree/main/otelzap with logging engine with use - Zap
+
+The discussion over this decision can be found in [celestia-node#663](https://github.com/celestiaorg/celestia-node/issues/663)
+and props to @liamsi for initial kickoff and a deep dive into OpenTelemetry.
+
+#### Tracing Backends
+
+For tracing, there are 3 modern OSS tools that are recommended. All of them have bidirectional support with OpenTelemetry:
+* [Uptrace](https://get.uptrace.dev/guide/#what-is-uptrace)
+  * The most recent (~1 year)
+  * The richest UI
+  * Tight to Clickhouse DB
+  * Made by OpenTelemetry
+  * The most lightweight
+* [Jaeger](https://www.jaegertracing.io/)
+  * The most mature
+  * Started by Uber, now supported by CNCF
+  * Supports multiple storages(ScyllaDB, InfluxDB, Amazon DynamoDB)
+* [Graphana Tempo](https://grafana.com/oss/tempo/)
+  * Deep integration with Graphana/Prometheus
+  * Relatively new (~2 years)
+  * Uses Azure, GCS, S3 or local disk for storage
+
+Each of these backends can be used independently and depending on the use case. For us, these are main use cases: 
+* Local development/debugging for the private network or even public network setup
+> I am personally planning to set up the lightweight Uptrace for the local light node. Just to play around and observe 
+> things
+* Data collection from the Testground test runs
+* Bootstrappers monitoring infrastructure
+* Data collection from the incentivized testnet participants
+
+There is no strict decision on which of these backends and where to use. People taking ownership of any listed vectors
+are free to use any recommended solution or any unlisted.
+
+#### Metrics Backend
+
+// WIP
+
+## Design
+
+### Tracing
+
+Tracing allows to see _how_ any process progresses through different modules, APIs and networks, as well as timings of 
+each operation and any events or errors as they occur.
+
+A visual example of a generic tracing dashboard provided via [Uptrace](https://uptrace.dev/) backend
+![tracing](img/tracing-dashboard.png)
+
+Mainly, for "ShrEx" and reconstruction analysis we need to know if the reconstruction succeeded and the time it took.
+The tracing in this case would provide all three metrics for the whole reconstruction operation and for each sub operation of each
```

Comment:
Explicitly mentioning what three metrics are can save the reader's time wondering. 
For e.g 
1 - reconstruction status
2 - reconstruction time
3 - ?

Reply:
I didn't made sub bullets here, but still improved and updated the paragraph with new info